backbone/src/main/java/org/researchstack/backbone/step/active/SensorRecorder.java

```diff
@@ -0,0 +1,148 @@
+package org.researchstack.backbone.step.active;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Handler;
+
+import org.researchstack.backbone.step.Step;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by TheMDP on 2/7/17.
+ *
+ * The SensorRecorder is an abstract class that greatly reduces the amount of work required
+ * to write sensor data to a DataLogger json file
+ *
+ * Any Android sensor is compatible with this class as long as you correctly implement
+ * the two abstract methods, getSensorTypeList, and writeJsonData
+ */
+
+abstract class SensorRecorder extends JsonArrayDataRecorder implements SensorEventListener {
+
+    private static final long MILLI_SECONDS_PER_SEC = 1000L;
+    private static final long MICRO_SECONDS_PER_SEC = 1000000L;
+
+    /**
+     * The frequency of accelerometer data collection in samples per second (Hz).
+     * Android Sensors do not allow exact frequency sepcifications, per their documentation,
+     * it is only a HINT, so we must manage it ourselves in a runnable here
+     */
+    private double frequency;
+
+    private SensorManager sensorManager;
+    private List<Sensor> sensorList;
+
+    /**
+     * the jsonObject that will be written to the file at frequency desired
+     */
+    private Handler mainHandler;
+    private Runnable jsonWriterRunnable;
+    private int  writeCounter;
+    private long writeDelayGoal;
+    private long writeStartTime;
+
+    /** Default constructor for serialization/deserialization */
+    SensorRecorder() {
+        super();
+    }
+
+    SensorRecorder(double frequency, String identifier, Step step, File outputDirectory) {
```

Comment:
The Recorders only exist for results that are too large to save in app memory, so non-time-series sensors don't really apply to the paradigm of SensorRecorder.  I think we would just make a separate StepLayout to accommodate and pull weight, blood pressure, etc.